deps/find_libpython.py

```diff
@@ -0,0 +1,376 @@
+    'python3.7m'
+    >>> library_name("libpython3.7m.so", suffix=".so", is_windows=False)
+    'python3.7m'
+    >>> library_name("libpython3.7m.dylib", suffix=".dylib", is_windows=False)
+    'python3.7m'
+    >>> library_name("python37.dll", suffix=".dll", is_windows=True)
+    'python37'
+    """
+    if not is_windows and name.startswith("lib"):
+        name = name[len("lib"):]
+    if suffix and name.endswith(suffix):
+        name = name[:-len(suffix)]
+    return name
+
+
+def append_truthy(list, item):
+    if item:
+        list.append(item)
+
+
+def uniquifying(items):
+    """
+    Yield items while excluding the duplicates and preserving the order.
+
+    >>> list(uniquifying([1, 2, 1, 2, 3]))
+    [1, 2, 3]
+    """
+    seen = set()
+    for x in items:
+        if x not in seen:
+            yield x
+        seen.add(x)
+
+
+def uniquified(func):
+    """ Wrap iterator returned from `func` by `uniquifying`. """
+    @functools.wraps(func)
+    def wrapper(*args, **kwds):
+        return uniquifying(func(*args, **kwds))
+    return wrapper
+
+
+@uniquified
+def candidate_names(suffix=SHLIB_SUFFIX):
+    """
+    Iterate over candidate file names of libpython.
+
+    Yields
+    ------
+    name : str
+        Candidate name libpython.
+    """
+    LDLIBRARY = sysconfig.get_config_var("LDLIBRARY")
+    if LDLIBRARY:
+        yield LDLIBRARY
+
+    LIBRARY = sysconfig.get_config_var("LIBRARY")
+    if LIBRARY:
+        yield os.path.splitext(LIBRARY)[0] + suffix
+
+    dlprefix = "" if is_windows else "lib"
+    sysdata = dict(
+        v=sys.version_info,
+        # VERSION is X.Y in Linux/macOS and XY in Windows:
+        VERSION=(sysconfig.get_config_var("VERSION") or
+                 "{v.major}.{v.minor}".format(v=sys.version_info)),
+        ABIFLAGS=(sysconfig.get_config_var("ABIFLAGS") or
+                  sysconfig.get_config_var("abiflags") or ""),
+    )
+
+    for stem in [
+            "python{VERSION}{ABIFLAGS}".format(**sysdata),
+            "python{VERSION}".format(**sysdata),
+            "python{v.major}".format(**sysdata),
+            "python",
+            ]:
+        yield dlprefix + stem + suffix
+
+
+
+@uniquified
+def candidate_paths(suffix=SHLIB_SUFFIX):
+    """
+    Iterate over candidate paths of libpython.
+
+    Yields
+    ------
+    path : str or None
+        Candidate path to libpython.  The path may not be a fullpath
+        and may not exist.
+    """
+
+    yield linked_libpython()
+
+    # List candidates for directories in which libpython may exist
+    lib_dirs = []
+    append_truthy(lib_dirs, sysconfig.get_config_var('LIBPL'))
+    append_truthy(lib_dirs, sysconfig.get_config_var('srcdir'))
+    append_truthy(lib_dirs, sysconfig.get_config_var("LIBDIR"))
+
+    # LIBPL seems to be the right config_var to use.  It is the one
+    # used in python-config when shared library is not enabled:
+    # https://github.com/python/cpython/blob/v3.7.0/Misc/python-config.in#L55-L57
```

Comment:
So it seems adding `LIBPL` was the key (it was not in the PyJulia version; I'll back-port it later).  This script finds libpython in case the old julia function couldn't: https://github.com/JuliaPy/PyCall.jl/issues/565#issuecomment-421340493